.github: Migrate iOS workflows to GHA

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#67645 .github: Migrate iOS workflows to GHA**

This PR migrates the iOS workflows to GHA from CircleCI. Also makes it so that iOS builds will run when `ciflow/ios` is added to a PR and will by default only run on `master`

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D32104367](https://our.internmc.facebook.com/intern/diff/D32104367)

seemethere committed:
.github: Migrate iOS workflows to GHA

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

janeyx99 reviewed (commented):
It seems that these three are run on PRs currently--are they now moved to trunk only?
![image](https://user-images.githubusercontent.com/31798555/139913328-46ab031e-2a85-475f-aaa6-fa48e343cc04.png)

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


This PR migrates the iOS workflows to GHA from CircleCI. Also makes it so that iOS builds will run when `ciflow/ios` is added to a PR and will by default only run on `master`

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D32104367](https://our.internmc.facebook.com/intern/diff/D32104367)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32104367).

seemethere commented on tools/lite_interpreter/gen_selected_mobile_ops_header.py:
```diff
@@ -127,7 +127,7 @@ def main() -> None:
     loaded_model = {}
     with open(model_file_name, "rb") as model_file:
-        loaded_model = yaml.load(model_file)
+        loaded_model = yaml.safe_load(model_file)
```
Changed to `safe_load` since `load` now leads a `Loader` argument

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


This PR migrates the iOS workflows to GHA from CircleCI. Also makes it so that iOS builds will run when `ciflow/ios` is added to a PR and will by default only run on `master`

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D32104367](https://our.internmc.facebook.com/intern/diff/D32104367)

[ghstack-poisoned]

seemethere commented:
Currently waiting on https://github.com/pytorch/pytorch/pull/67694, will rebase after that gets merged

seemethere commented:
> It seems that these three are run on PRs currently--are they now moved to trunk only? ![image](https://user-images.githubusercontent.com/31798555/139913328-46ab031e-2a85-475f-aaa6-fa48e343cc04.png)

Yes considering that the iOS builds don't fail often, rely on expensive / slow executors (macOS), and are specialized in general I think it's pretty safe to move them to trunk only and let iOS PyTorch devs set the default ciflow for them to include `ciflow/ios`

seemethere committed:
Update on ".github: Migrate iOS workflows to GHA"


This PR migrates the iOS workflows to GHA from CircleCI. Also makes it so that iOS builds will run when `ciflow/ios` is added to a PR and will by default only run on `master`

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D32104367](https://our.internmc.facebook.com/intern/diff/D32104367)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32104367).
